src/main/java/org/team1294/firstpowerup/robot/subsystems/VisionSubsystem.java

```diff
@@ -0,0 +1,191 @@
+package org.team1294.firstpowerup.robot.subsystems;
+
+import edu.wpi.cscore.CvSink;
+import edu.wpi.cscore.CvSource;
+import edu.wpi.cscore.UsbCamera;
+import edu.wpi.first.wpilibj.CameraServer;
+import edu.wpi.first.wpilibj.command.Subsystem;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import org.opencv.core.Mat;
+import org.opencv.core.Rect;
+import org.opencv.core.Scalar;
+import org.opencv.imgproc.Imgproc;
+import org.team1294.firstpowerup.robot.Robot;
+import org.team1294.firstpowerup.robot.commands.DefaultVisionCommand;
+import org.team1294.firstpowerup.robot.vision.CratePipeline;
+import org.team1294.firstpowerup.robot.vision.PairOfRect;
+import org.team1294.firstpowerup.robot.vision.SwitchTargetPipeline;
+import org.team1294.firstpowerup.robot.vision.VisionProcessingResult;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class VisionSubsystem extends Subsystem {
+
+    private static final int IMG_WIDTH = 320;
+    private static final int IMG_HEIGHT = 240;
+    private final CvSink cvSink;
+    private final CvSource cvSource;
+    private final UsbCamera usbCamera;
+    private final CameraServer cameraServer;
+    private final Mat frame = new Mat();
+    private final SwitchTargetPipeline switchTargetPipeline;
+    private final CratePipeline cratePipeline;
+
+    private VisionProcessingResult visionProcessingResult;
+
+    public VisionSubsystem() {
+        switchTargetPipeline = new SwitchTargetPipeline();
+        cratePipeline = new CratePipeline();
+
+        cameraServer = CameraServer.getInstance();
+
+        usbCamera = cameraServer.startAutomaticCapture(0);
+        usbCamera.setResolution(IMG_WIDTH, IMG_HEIGHT);
+        usbCamera.setFPS(30);
```

Comment:
Probably should be a constant like resolution.

Reply:
consider it done